lib/OpenQA/Base.pm

```diff
@@ -13,20 +13,110 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# This package contains shared functions between WebAPI and WebSockets
-package OpenQA::ServerStartup;
-
+package OpenQA::Base;
+use Mojo::Log;
+use Mojo::Home;
 use strict;
 use warnings;
+use Mojo::Base -base;
+use Sys::Hostname;
+use File::Spec::Functions 'catfile';
+use Mojo::File 'path';
 use Config::IniFiles;
 use db_profiler;
 use db_helpers;
 use OpenQA::Utils;
-use Mojo::File 'path';
 
-sub read_config {
-    my $app = shift;
+has config => sub { {} };
+
+has log => sub { setup_log(shift) };
+
+has home => sub { Mojo::Home->new($ENV{MOJO_HOME} || '/') };
+
+has mode => 'production';
+
+has log_name => 'scheduler';
+
+has level => 'debug';
+
+has instance => undef;
+
+has log_dir => undef;
+
+has schema => sub { OpenQA::Schema::connect_db() };
+
+sub setup_log {
+    my ($self) = @_;
+    my ($logfile, $logdir, $level, $log);
+
+    if ($self->isa('OpenQA::Base')) {
```

Comment:
Why the rename to `OpenQA::Base`?, or did you wanted to use some Inheritance so you would have the server startup part in ResourceAllocator, Scheduler, Websockets and Server?... If that's the case, you can create a base class that only has the log configuration and stuff and `use base 'ClassName'` allowing to benefit from inheritance

Reply:
I did the renaming because of coolo's comment:

>  coolo 6 days ago Contributor
>
> Can we not spread the name FakeApp then?

Reply:
I see, but that rename from OpenQA::ServerStartup to ::Base makes a call use base 'OpenQA::Base' a bit odd if one wishes to create a new service. (Although, +1 for removing FakeApp :P)... Mind changing back to ServerStartup?